tests/rptest/tests/self_test_test.py

```diff
@@ -233,3 +233,66 @@ def test_self_test_cancellable(self):
                 # cancelled before it even had a chance to start, resulting in
                 # a 0 value for duration
                 assert report['duration'] >= 0
+
+    @cluster(num_nodes=3)
+    def test_self_test_unknown_test_type(self):
+        """Assert the self test still runs when an invalid test type is passed.
+           This helps ensure the self test will still work in a cluster
+           with mixed versions of Redpanda."""
+        num_nodes = 3
+
+        self.start_redpanda(num_nodes=num_nodes,
+                            si_settings=SISettings(
+                                test_context=self.test_context,
+                                cloud_storage_enable_remote_read=True,
+                                cloud_storage_enable_remote_write=True))
+
+        #Attempt to run with an unknown test type "pandatest"
+        #and possibly unknown "cloud" test.
```

Comment:
nit: would be good to be consistent about spacing, even in comments (#<space>comment)

Reply:
Will fix in follow up.